selectors/views/channel_sidebar.ts

```diff
@@ -18,3 +24,37 @@ export function isCategoryCollapsed(state: GlobalState, categoryId: string) {
 export function isUnreadFilterEnabled(state: GlobalState) {
     return state.views.channelSidebar.unreadFilterEnabled;
 }
+
+export function makeGetCurrentlyDisplayedChannelsForTeam(): (state: GlobalState, teamId: string) => Channel[] {
+    const getCategoriesForTeam = makeGetCategoriesForTeam();
+    const getChannelsByCategory = makeGetChannelsByCategory();
+
+    return createSelector(
+        (state: GlobalState) => state,
```

Comment:
As discussed, this selector won't work because we have to pass the whole state into the result function which invalidates the whole thing.